shared/src/TwinCollection.cs

```diff
@@ -40,7 +40,7 @@ public TwinCollection()
         /// </summary>
         /// <param name="twinJson">JSON fragment containing the twin data.</param>
         public TwinCollection(string twinJson)
-            : this(JObject.Parse(twinJson))
+            : this(JsonConvert.DeserializeObject<JObject>(twinJson))
```

Comment:
By changing here and below, we can take advantage of the internal serializer settings we added before and avoid of the unexpected behavior of dropping trailing zeros (2023-01-31T10:37:08.45994**00** -> 2023-01-31T10:37:08.45994)

Reply:
I'd recommend adding a comment that explains this implementation choice.